fix for voq fabric related tests

rawal01 opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/sonic-net/SONiC/blob/gh-pages/CONTRIBUTING.md

Please provide following information to help code review process a bit easier:
-->
### Description of PR
<!--
- Please include a summary of the change and which issue is fixed.
- Please also include relevant motivation and context. Where should reviewer start? background context?
- List any dependencies that are required for this change.
-->

Summary:
Fixes # 16067 
https://github.com/sonic-net/sonic-mgmt/issues/16067

### Type of change

<!--
- Fill x for your type of change.
- e.g.
- [x] Bug fix
-->

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x ] Test case(new/improvement)


### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [X ] 202405

### Approach
#### What is the motivation for this PR?
Fixing issue 16067. 
#### How did you do it?
Changed to not re-write originalIsolateStatus but instead write to isolateStatus.
Also added try/catch block to both tests as today if the test fails it exits without restoring the state of the port.

I also added check  to tests/voq/test_fabric_reach.py::test_fabric_reach_supervisor to run against only if asic is present. Right not the tests get the max asics for the platform and assumes they are all present.
#### How did you verify/test it?
Ran against t2 Nokia chassis
#### Any platform specific information?

#### Supported testbed topology if it's a new test case?

### Documentation
<!--
(If it's a new feature, new test case)
Did you update documentation/Wiki relevant to your implementation?
Link to the wiki page?
-->

mssonicbld commented:
/azp run

rawal01 commented:
@mssonicbld I checked the test failed because of host unreachable in posttest not related to any changes I made please rerun

arlakshm commented:
@jfeng-arista, can you review this change

mssonicbld commented:
/azp run

mssonicbld commented:
/azp run

kenneth-arista commented on tests/voq/test_fabric_cli_and_db.py:
```diff
@@ -172,122 +177,127 @@ def test_fabric_cli_isolate_supervisor(duthosts, enum_supervisor_dut_hostname):
+                                                                                                          localPort)
+                    cmd_output = duthost.shell(cmd, module_ignore_errors=True)["stdout"].split("\n")
+                    tokens = cmd_output[0].split()
+                    isolateStatus = tokens[0]
```
I agree with the change to not change `originalIsolateStatus`. So removing the original line makes sense. However, `isolateStatus` should be set the pytest_assert() check.

kenneth-arista commented on tests/voq/test_fabric_cli_and_db.py:
```diff
@@ -172,122 +177,127 @@ def test_fabric_cli_isolate_supervisor(duthosts, enum_supervisor_dut_hostname):
+                                                                                                              localPort)
+                    cmd_output = duthost.shell(cmd, module_ignore_errors=True)["stdout"].split("\n")
+                    tokens = cmd_output[0].split()
+                    isolateStatus = tokens[0]
```
Same comment as above. Can you move this to after the `pytest_assert`?

kenneth-arista commented on tests/voq/test_fabric_cli_and_db.py:
```diff
@@ -172,122 +177,127 @@ def test_fabric_cli_isolate_supervisor(duthosts, enum_supervisor_dut_hostname):
+                    cmd_output = duthost.shell(cmd, module_ignore_errors=True)["stdout"].split("\n")
+                    tokens = cmd_output[0].split()
+                    isolateStatus = tokens[0]
+                    pytest_assert(len(tokens) > 0,
```
Same comment as above. Can you move this `ptest_assert` to above access `tokens[0]`?

kenneth-arista commented on tests/voq/test_fabric_cli_and_db.py:
```diff
@@ -172,122 +177,127 @@ def test_fabric_cli_isolate_supervisor(duthosts, enum_supervisor_dut_hostname):
+                    cmd_output = duthost.shell(cmd, module_ignore_errors=True)["stdout"].split("\n")
+                    tokens = cmd_output[0].split()
+                    isolateStatus = tokens[0]
+                    pytest_assert(
```
Same comment as above. Please move `pytest_assert` to above accessing `tokens[0]`

kenneth-arista reviewed (approved):
Besides the minor comments, this change looks good to me.

mssonicbld commented:
/azp run

mssonicbld commented:
/azp run

rawal01 commented:
@arlakshm can you help merge this PR ?

jfeng-arista commented:
The change looks good to me, also quick tested on one of our systems and they are running ok.  We can merge this up after Kenneth's comments get addressed.

rawal01 commented:
> The change looks good to me, also quickly tested on one of our systems and they are running ok. We can merge this up after Kenneth's comments get addressed.

@jfeng-arista  I already addressed all comments in latest commit  please check

rlhui commented:
@bingwang-ms  please approve? thanks.

mssonicbld commented:
Cherry-pick PR to 202405: https://github.com/sonic-net/sonic-mgmt/pull/16393

jfeng-arista commented:
> > The change looks good to me, also quickly tested on one of our systems and they are running ok. We can merge this up after Kenneth's comments get addressed.
> 
> @jfeng-arista I already addressed all comments in latest commit please check

great ! thank you
